spec.bs

```diff
@@ -836,50 +922,52 @@ On the other hand, methods for getting data from the [=shared storage database=]
-            1. If |resultURLWithMetadata| has field "`reportingMetadata`", run [=register reporting metadata=] with |resultURLWithMetadata|["`reportingMetadata`"].
-        1. [=Upon rejection=] of |indexPromise|, set |fencedFrameConfigStruct|'s [=/url=] to |urlList|[[=default index=]].
+    1. Let |indexPromise| be the result of running [=get the select-url result index=], given |worklet|, |name|, |urlList|, and |options|.
+    1. [=Upon fulfillment=] of |indexPromise|, perform the following steps:
+        1. Let |resultIndex| be the numerical value of |indexPromise|.
+        1. Let |context| be {{WindowSharedStorage}}'s {{Window}}'s [=Window/browsing context=].
+        1. If |context| is null, let |resultURLWithMetadata| be |urls|[[=default index=]] and abort these steps.
+        1. Let |document| be |context|'s [=active document=].
+        1. If |document| is not [=fully active=], let |resultURLWithMetadata| be |urls|[[=default index=]] and abort these steps.
+        1. Let |origin| be |document|'s [=url/origin=].
+        1. [=Assert=] that |origin| is not [=opaque origin|opaque=].
+        1. Let |remainingBudget| be the result of running [=determine remaining navigation budget=] with |environment| and |origin|.
+        1. [=Assert=] that |remainingBudget| is not undefined.
+        1. Let |pendingBits| be the logarithm base 2 of |urlList|'s [=list/size=].
+        1. If |pendingBits| is greather than |remainingBudget|, set |resultIndex| to [=default index=].
+        1. Set |fencedFrameConfigStruct|'s [=pending shared storage budget debit=] to |pendingBits|.
+        1. Set |fencedFrameConfigStruct|'s [=/url=] to |urlList|[|resultIndex|].
+        1. Let |resultURLWithMetadata| be |urls|[|resultIndex|].
+        1. If |resultURLWithMetadata| has field "`reportingMetadata`", run [=register reporting metadata=] with |resultURLWithMetadata|["`reportingMetadata`"].
+        1. If |options|["`keepAlive`"] is false, run [=terminate a worklet global scope=] with the associated {{SharedStorageWorkletGlobalScope}}.
+    1. [=Upon rejection=] of |indexPromise|, perform the following steps:
+        1. Set |fencedFrameConfigStruct|'s [=/url=] to |urlList|[[=default index=]].
+        1. If |options|["`keepAlive`"] is false, run [=terminate a worklet global scope=] with the associated {{SharedStorageWorkletGlobalScope}}.
     1. Return |resultPromise|.
-    1. If |options|["`keepAlive`"] is false, [=upon fulfillment=] of |resultPromise| or [=upon rejection=] of |resultPromise|, run [=terminate a worklet global scope=] with the associated {{SharedStorageWorkletGlobalScope}}.
+  </div>
+
+  ## Extension to the {{Window}} interface ## {#extension}
+
+  Each {{Window}} object has an associated {{WindowSharedStorage}} instance {{Window/sharedStorage}}, which is created alongside the .{{Window}} if [=Shared Storage=] is enabled, with the [=Window/sharedStorage getter|getter=] below.
```

Comment:
Can you clarify what "is enabled" means? Does this getter only exist if a permission policy is enabled (and we can then link to the "allowed" algorithm dfn?) Or does this mean something else..